game-server-hosting/server/event_watcher.go

```diff
@@ -64,6 +68,10 @@ func (s *Server) watchAllocation(ev localproxy.Event) {
 // channel when signalled.
 func (s *Server) watchDeallocation(ev localproxy.Event) {
 	if de, ok := ev.(*localproxy.DeallocateEvent); ok {
+		s.allocatedUUIDMtx.Lock()
+		s.allocatedUUID = de.AllocationID
```

Comment:
```suggestion
		s.allocatedUUID = ""
```

Deallocation needs to unset the ID.